### ISSUE TITLE
Dodge can occur multiple times without any dodge modifiers

### ISSUE THREAD

**Issue by Siersha** [CleverRaven/Cataclysm-DDA#27203]
**Describe the bug**  
A player may dodge up to an infinite number of times without the actual use of zui quan or any other dodges_left changing martial art. Speculatively, it's because of the lack of any reference to dodges_left in melee.cpp outside of the dodge reduction for every consecutive dodge (which is also a problem area). Since dodges are calculated off of dodges_left, when it is less than 0 (and still allowing dodges), it gives you infinite dodges of dodge level - 2 skill.
 [https://github.com/CleverRaven/Cataclysm-DDA/blob/e50236488f73bf854cd5fabffdbcd93a9477834d/src/melee.cpp](url) 

**To Reproduce**  
1. make a player with high dodge (either debug skill or very high dex).
2. surround yourself with enemies with relatively low accuracy (zombie child for example).
3. stand still and watch yourself dodge many attacks within 1 turn as if you had zui-quan dodge bonus.

**Expected behavior**  
You dodge only once outside of special martial arts bonuses.

**Comment by ZhilkinSerg** [comment]
Duplicate of #27123 probably.

**Comment by Siersha** [comment]
> Duplicate of #27123 probably.

They're referencing two completely different bugs. One is on the implementation of dodge counters (upon attacking), and this one is on the implementation of the number of dodges allowed outside of martial arts.
